addons/l10n_es_edi_facturae/models/account_move_send.py

```diff
@@ -0,0 +1,458 @@
+                'name': filename,
+                'mimetype': 'application/xml',
+                'placeholder': True,
+            })
+
+        return results
+
+    # -------------------------------------------------------------------------
+    # BUSINESS ACTIONS
+    # -------------------------------------------------------------------------
+
+    def _prepare_document(self, invoice):
+        # EXTENDS 'account'
+        results = super()._prepare_document(invoice)
+
+        if self.l10n_es_edi_facturae_checkbox_xml and self._l10n_es_edi_facturae_get_default_enable(invoice):
+            try:
+                xml_content = self._l10n_es_edi_facturae_render_facturae(invoice)
+            except Exception as error:
+                return {
+                    'error': "".join((_("Errors occured while creating the EDI document (format: %s):", "Facturae"),
+                        "\n", str(error))),
+                }
+
+            results['l10n_es_edi_facturae_attachment_values'] = {
+                'name': f'{invoice.name.replace("/", "_")}_facturae_signed.xml',
+                'raw': xml_content,
+                'mimetype': 'application/xml',
+                'res_model': invoice._name,
+                'res_id': invoice.id,
+                'res_field': 'l10n_es_edi_facturae_xml_file',  # Binary field
+            }
+
+        return results
+
+    def _link_document(self, invoice, prepared_data):
+        # EXTENDS 'account'
+        super()._link_document(invoice, prepared_data)
+
+        attachment_vals = prepared_data.get('l10n_es_edi_facturae_attachment_values')
+        if attachment_vals:
+            self.env['ir.attachment'].create(attachment_vals)
+            invoice.invalidate_model(fnames=['l10n_es_edi_facturae_xml_id', 'l10n_es_edi_facturae_xml_file'])
+
+
+    # -------------------------------------------------------------------------
+    # EDI
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _l10n_es_edi_facturae_inv_lines_to_items(self, invoice):
+        """
+        Convert the invoice lines to a list of items required for the Facturae xml generation
+
+        :param account.move invoice:    The invoice containing the invoice lines
+        :return:                        A tuple containing the Face items, the taxes and the invoice totals data.
+        """
+        items = []
+        totals = {
+            'total_gross_amount': 0.,
+            'total_general_discounts': 0.,
+            'total_general_surcharges': 0.,
+            'total_taxes_withheld': 0.,
+            'total_tax_outputs': 0.,
+            'total_payments_on_account': 0.,
+            'amounts_withheld': 0.,
+        }
+        taxes = []
+        taxes_withheld = []
+        for line in invoice.invoice_line_ids:
+            invoice_line_values = {}
+
+            discount = max(0., line.discount * (line.discount + line.price_subtotal) / 100)
```

Comment:
???
If price_unit = 1000 with 10% discount:
`discount = max(0, 10 * (10 + 1000) / 100) = 101`
It's completely wrong, right?